src/bindings/python/src/pyopenvino/utils/utils.cpp

```diff
@@ -449,8 +452,20 @@ ov::Any py_object_to_any(const py::object& py_obj) {
         return py::cast<ov::intel_auto::SchedulePolicy>(py_obj);
     } else if (py::isinstance<ov::hint::SchedulingCoreType>(py_obj)) {
         return py::cast<ov::hint::SchedulingCoreType>(py_obj);
-    } else if (py::isinstance<std::set<ov::hint::ModelDistributionPolicy>>(py_obj)) {
-        return py::cast<std::set<ov::hint::ModelDistributionPolicy>>(py_obj);
+    } else if (py::isinstance<py::set>(py_obj)) {
+        auto _set = py_obj.cast<py::set>();
+
+        PY_TYPE detected_type = check_list_element_type<py::set>(_set);
+
+        if (_set.empty())
+            return ov::Any(EmptyList());
+
+        switch (detected_type) {
+        case PY_TYPE::ModelDistributionPolicy:
```

Comment:
I think it also good for python api to add other simple cases here like int, float, bool  and str :)
And cover by tests